datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -3803,7 +3807,7 @@ pub(crate) mod tests {
             "SortPreservingMergeExec: [c@2 ASC]",
             // Expect repartition on the input to the sort (as it can benefit from additional parallelism)
             "SortExec: expr=[c@2 ASC]",
-            "ProjectionExec: expr=[a@0 as a]",
```

Comment:
Previously after projection there was no column c. However, sort above refers to c, this example changes projection below so that column c is still present at the input of the Sort (All of the changes in this PR are in similar spirit).